aflr/api_request.py

```diff
@@ -4,52 +4,72 @@
 import os
 import requests
 from requests.exceptions import HTTPError
+from functools import lru_cache
 
 
 class APIRequest:
-    def __init__(self, api_key=None, api_base=None, **params):
-        self.api_key = api_key if api_key else aflr.api_key
-        self.api_base = api_base if api_base else aflr.api_base
-        if self.api_key == None or self.api_key == "your-key":
-            self.api_key = os.environ.get("aflr_key", None)
-        if self.api_key == None or len(self.api_key) < 30:
+    @lru_cache(maxsize=10, typed=False)
```

Comment:
why 10? why not a bigger number

Reply:
api key should not change often (or change at all normally). I think 10 is enough for cache

Reply:
removed memo